recipe-client-addon/test/browser/Utils.jsm

```diff
@@ -27,4 +27,48 @@ this.Utils = {
       yield testGenerator(driver);
     });
   },
+
+  /**
+   * Test wrapper that replaces a property on an object with another value for
+   * the duration of the test. The test generator is passed the replaced value.
+   */
+  withPatch(object, propertyName, patchValue, testGenerator) {
+    return function* inner() {
+      const original = object[propertyName];
+      object[propertyName] = patchValue;
+      yield testGenerator(patchValue);
+      object[propertyName] = original;
+    };
+  },
+
+  /**
+   * Creates a wrapper function that tracks whether it has been called.
+   */
+  createMock(wrapped) {
```

Comment:
I think other people have used `sinon` for this and similar tasks. I'm not super familiar with it, but at least for this particular util I believe `sinon.spy()` does what you want. See https://dxr.mozilla.org/mozilla-central/source/browser/components/preferences/in-content/tests/browser_advanced_siteData.js#9-10 (and further down) for an example.